### PR TITLE
feat: zod validator

### PR DESCRIPTION
**Pull request by Asuka109**
# PR Details

## Description

image

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My code follows the code style of this project.
- [ ] I have updated changeset
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: packages/builder/builder-webpack-provider/scripts/post-compile.js [hunk 1]
```diff
@@ -1,18 +1,8 @@
-const { ConfigValidator } = require('../dist/config/validate');
 const path = require('path');
 const { readFile, writeFile } = require('fs/promises');
 const { transformAsync } = require('@babel/core');
 const { performance } = require('perf_hooks');
 
-async function compileAjv() {
-  const output = path.resolve(
-    __dirname,
-    '../compiled/config-validator/index.js',
```
Comment (chenjiahan) [inline packages/builder/builder-webpack-provider/scripts/post-compile.js:10]: compiled/config-validator 这个文件可以删了

Comment (Asuka109) [reply packages/builder/builder-webpack-provider/scripts/post-compile.js:10]: okk

File: packages/toolkit/utils/package.json [hunk 2]
```diff
@@ -178,16 +178,17 @@
   },
   "dependencies": {
     "caniuse-lite": "^1.0.30001332",
-    "lodash": "^4.17.21"
+    "lodash": "^4.17.21",
+    "zod": "^3.17.3"
```
Comment (chenjiahan) [inline packages/toolkit/utils/package.json:182]: 这里 zod 似乎还没用到

File: packages/builder/builder-webpack-provider/src/config/validate/tools.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+import { z } from '@modern-js/builder-shared';
+import type { ToolsConfig } from '../../types';
+
+export const toolsConfigSchema: z.ZodType<ToolsConfig> = z.partialObj({
+  pug: z.union([z.literal(true), z.chained(z.any(), z.any())]),
```
Comment (chenjiahan) [inline packages/builder/builder-webpack-provider/src/config/validate/tools.ts:5]: 为啥是 any

Comment (Asuka109) [reply packages/builder/builder-webpack-provider/src/config/validate/tools.ts:5]: pug options 作为传给第三方库的 config 感觉就不用在这里做校验了，所以只做了 chained config 的形态校验

Comment (Asuka109) [reply packages/builder/builder-webpack-provider/src/config/validate/tools.ts:5]: 我才发现这个地方 tools config 的其它字段还没定义，等我补下..

**Review by chenjiahan — APPROVED** [review]

**Pull request merged**